src/components/SideMenu.astro

```diff
@@ -5,14 +5,17 @@ import { app } from "../lib/firebase_server.ts";
 /* Check if the user is authenticated */
 const auth = getAuth(app);
 let userEmail = "[Display User Email]";
+let userName = "[Full Name]";
 
 if (Astro.cookies.has("__session")) {
   const sessionCookie = Astro.cookies.get("__session")?.value;
   if (sessionCookie) {
     try {
       const decodedCookie = await auth.verifySessionCookie(sessionCookie);
-      if (decodedCookie.email) {
-        userEmail = decodedCookie.email;
+      const user = await auth.getUser(decodedCookie.uid);
+      if (user) {
+        //userEmail = decodedCookie.email;
+        userName = user.displayName || user.email || "Fail to retrieve user name";
```

Comment:
Finally I had a good idea: display the `uid` (guaranteed to be defined) in the fallback case.